File: pkg/rbac/models/role.go
package models

type Role struct {
	ID      int            `json:"id"`
	Name    string         `json:"name"`
	Entries []FeatureEntry `json:"entries"`
}

type FeatureEntry struct {
	ID          int        `json:"id,omitempty" yaml:"entries,omitempty"`
	Name        string     `json:"name"`
	Description string     `json:"description"`
	Endpoints   []Endpoint `json:"endpoints"`
}

type Endpoint struct {
	ID     int    `json:"id,omitempty" yaml:"entries,omitempty"`
	Name   string `json:"name"`
	Path   string `json:"path"`
	Method string `json:"method"`
}

type RoleTmpl struct {
	Entries []FeatureEntry `json:"entries"`
}

File: ui/src/app/models/role.ts
export interface Endpoint {
  id: number;
  name: string;
  path: string;
  method: string;
}

export interface FeatureEntry {
  id: number;
  name: string;
  description: string;
  endpoints: Endpoint[];
}

export interface Role {
  id: number;
  name: string;
  entries: FeatureEntry[];
}
